lib/Service/Attachment/AttachmentService.php

```diff
@@ -249,6 +259,31 @@ public function handleAttachments(Account $account, array $attachments, \Horde_I
 		return array_values(array_filter($attachmentIds));
 	}
 
+	public function getAttachmentNames(Account $account, Mailbox $mailbox, Message $message, \Horde_Imap_Client_Socket $client): array {
+		$attachments = [];
+		$cached = $this->cache->get($message->getUid());
+		if ($cached) {
+			return $cached;
+		}
+		try {
+			$imapMessage = $this->mailManager->getImapMessage(
+				$client,
+				$account,
+				$mailbox,
+				$message->getUid(),
+				true
+			);
+			$attachments = $imapMessage->getAttachments();
+		} catch (ServiceException $e) {
+			$this->logger->error('Could not get attachment names', ['exception' => $e, 'messageId' => $message->getUid()]);
+		}
+		$result = array_map(static function (array $attachment) {
+			return $attachment['fileName'];
+		}, $attachments);
+		$this->cache->set($message->getUid(), $result);
```

Comment:
UIDs are unique in the scope of a single mailbox. You'll run into conflicts across mailboxes, and worse, across user accounts. Use the primary key or find another unique identifier.